<font color='red' size='6'>REMINDER</font>

# Fill in any place that says `YOUR CODE HERE`.
- You should remove the line that says `raise NotImplementedError()`. If you do not, your code will (unsurprisingly) throw a run-time error and cause everything to fail.
- Do **NOT** write your answer anywhere else other than where it says `YOUR CODE HERE`. Simply write your code directly below this comment in the **same code cell**.

# Make sure everything runs as expected.
- Go to the menubar, select *Kernel* > *Restart & Run All*

# Do <ins>NOT</ins> change the title (i.e., file name) of this notebook.

# Do <ins>NOT</ins> delete any of the cells in this notebook.

# Make sure you save your work
- Go to the menubar, select *File* > *Save and Checkpoint*

----

# Problem Description 

You have been contracted to help a US-based hotel chain with some data analysis. They have provided you historical survey results and want to be able to use the data to help predict someone's intent to recommend the hotel to their friends. The file can be found in the `data` subfolder and is named `hotel_survey.xlsx`.

The data dictionary is as follows:
- Record ID: a unique identifier for the survey response
- Arrival Date: the date when the person stayed at the hotel
- Loyalty Program Tier: the patron's loyalty tier 
- Age: the age, in years, of the patron
- Gender: the gender of the patron
- Maintenance and Upkeep: a rating between 1 and 10 of how the patron perceived the hotel's maintenance and upkeep
- Staff Service Overall: a rating between 1 and 10 of how the patron perceived the overall service from the hotel staff
- Overall Satisfaction: a rating between 1 and 10 of the patron's overall satisfication with their stay
- Intent To Recommend: a rating between 1 and 10 of how likely the patron will recommend the hotel to their friends

For all of the ratings, a higher number is better.

Complete the following steps for your analysis.

In [1]:
# Run this code cell 
from nose.tools import assert_equal, assert_in, assert_is_instance
from nose.tools import assert_almost_equal, assert_not_equal

## Step 1

Read the data from the file into a `DataFrame` named `hotel`.

In [2]:
# YOUR CODE HERE
import pandas as pd

# read in the file
hotel = pd.read_excel('./data/hotel_survey.xlsx')

In [3]:
# view hotel and info
hotel

Record ID Arrival Date Loyalty Program Tier  Age  Gender  \
0             1   2019-02-08                Basic   92  Female   
1             2   2019-05-26               Silver   58  Female   
2             3   2019-10-12           Non-Member   81  Female   
3             4   2019-09-30           Non-Member   68    Male   
4             5   2019-09-16                Basic   54  Female   
...         ...          ...                  ...  ...     ...   
8467       8468   2019-07-26           Non-Member   70  Female   
8468       8469   2019-09-30             Platinum   68    Male   
8469       8470   2019-11-25                Basic   45    Male   
8470       8471   2019-03-19           Non-Member   38  Female   
8471       8472   2019-07-02             Platinum   69    Male   

      Maintenance and Upkeep  Staff Service Overall  Overall Satisfaction  \
0                          8                      5                     7   
1                          8                      5                     6   
2                         10                      8                     5   
3                          7                      6                     5   
4                         10                      8                     7   
...                      ...                    ...                   ...   
8467                      10                      9                    10   
8468                       9                      8                     5   
8469                       8                      5                     5   
8470                       8                      9                     8   
8471                      10                      9                    10   

      Intent To Recommend  
0                       5  
1                       5  
2                       4  
3                       4  
4                       9  
...                   ...  
8467                   10  
8468                    7  
8469                    3  
8470                    8  
8471                    8  

[8472 rows x 9 columns]

In [4]:
# hotel info
hotel.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8472 entries, 0 to 8471
Data columns (total 9 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   Record ID               8472 non-null   int64         
 1   Arrival Date            8472 non-null   datetime64[ns]
 2   Loyalty Program Tier    8472 non-null   object        
 3   Age                     8472 non-null   int64         
 4   Gender                  8472 non-null   object        
 5   Maintenance and Upkeep  8472 non-null   int64         
 6   Staff Service Overall   8472 non-null   int64         
 7   Overall Satisfaction    8472 non-null   int64         
 8   Intent To Recommend     8472 non-null   int64         
dtypes: datetime64[ns](1), int64(6), object(2)
memory usage: 595.8+ KB


In [5]:
# This is a test cell
# If **NO** message is printed, it means that the tests passed
# One basic test to see if your code works
assert_equal(hotel.shape[0], 8472, msg='You do not have the correct number of rows')
assert_equal(hotel.shape[1], 9, msg='You do not have the correct number of columns')

## Step 2
Change all the column names in `hotel` to be lower case and replace all spaces with underscores.

In [6]:
# YOUR CODE HERE
# lowercase
hotel.columns = hotel.columns.str.lower()

In [7]:
# confirm
hotel.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8472 entries, 0 to 8471
Data columns (total 9 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   record id               8472 non-null   int64         
 1   arrival date            8472 non-null   datetime64[ns]
 2   loyalty program tier    8472 non-null   object        
 3   age                     8472 non-null   int64         
 4   gender                  8472 non-null   object        
 5   maintenance and upkeep  8472 non-null   int64         
 6   staff service overall   8472 non-null   int64         
 7   overall satisfaction    8472 non-null   int64         
 8   intent to recommend     8472 non-null   int64         
dtypes: datetime64[ns](1), int64(6), object(2)
memory usage: 595.8+ KB


In [8]:
# remove spaces and add underscore
hotel.rename(columns={'record id': 'record_id', 'arrival date': 'arrival_date', 'loyalty program tier': 'loyalty_program_tier',
                     'maintenance and upkeep' : 'maintenance_and_upkeep', 'staff service overall' : 'staff_service_overall',
                     'overall satisfaction' : 'overall_satisfaction', 'intent to recommend' : 'intent_to_recommend'}, inplace=True)

In [9]:
# confirm all lowercase and all spaces replaced with underscores
hotel.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8472 entries, 0 to 8471
Data columns (total 9 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   record_id               8472 non-null   int64         
 1   arrival_date            8472 non-null   datetime64[ns]
 2   loyalty_program_tier    8472 non-null   object        
 3   age                     8472 non-null   int64         
 4   gender                  8472 non-null   object        
 5   maintenance_and_upkeep  8472 non-null   int64         
 6   staff_service_overall   8472 non-null   int64         
 7   overall_satisfaction    8472 non-null   int64         
 8   intent_to_recommend     8472 non-null   int64         
dtypes: datetime64[ns](1), int64(6), object(2)
memory usage: 595.8+ KB


In [10]:
# This is a test cell
# If **NO** message is printed, it means that the tests passed
# One basic test to see if your code works
assert_equal(hotel.columns[5], 'maintenance_and_upkeep', msg='You did not replace the column names correctly')

<hr>

## Step 3
Create dummy variables for the categorical columns being sure to use the argument `dtype=int` to make the values in the new columns 0 or 1. Store the resulting `DataFrame` in a variable named `hotel_coded`.

In [11]:
# YOUR CODE HERE
hotel_coded = pd.get_dummies(data=hotel, dtype=int)

In [12]:
# view
hotel_coded

record_id arrival_date  age  maintenance_and_upkeep  \
0             1   2019-02-08   92                       8   
1             2   2019-05-26   58                       8   
2             3   2019-10-12   81                      10   
3             4   2019-09-30   68                       7   
4             5   2019-09-16   54                      10   
...         ...          ...  ...                     ...   
8467       8468   2019-07-26   70                      10   
8468       8469   2019-09-30   68                       9   
8469       8470   2019-11-25   45                       8   
8470       8471   2019-03-19   38                       8   
8471       8472   2019-07-02   69                      10   

      staff_service_overall  overall_satisfaction  intent_to_recommend  \
0                         5                     7                    5   
1                         5                     6                    5   
2                         8                     5                    4   
3                         6                     5                    4   
4                         8                     7                    9   
...                     ...                   ...                  ...   
8467                      9                    10                   10   
8468                      8                     5                    7   
8469                      5                     5                    3   
8470                      9                     8                    8   
8471                      9                    10                    8   

      loyalty_program_tier_Basic  loyalty_program_tier_Gold  \
0                              1                          0   
1                              0                          0   
2                              0                          0   
3                              0                          0   
4                              1                          0   
...                          ...                        ...   
8467                           0                          0   
8468                           0                          0   
8469                           1                          0   
8470                           0                          0   
8471                           0                          0   

      loyalty_program_tier_Non-Member  loyalty_program_tier_Platinum  \
0                                   0                              0   
1                                   0                              0   
2                                   1                              0   
3                                   1                              0   
4                                   0                              0   
...                               ...                            ...   
8467                                1                              0   
8468                                0                              1   
8469                                0                              0   
8470                                1                              0   
8471                                0                              1   

      loyalty_program_tier_Silver  gender_Female  gender_Male  
0                               0              1            0  
1                               1              1            0  
2                               0              1            0  
3                               0              0            1  
4                               0              1            0  
...                           ...            ...          ...  
8467                            0              1            0  
8468                            0              0            1  
8469                            0              0            1  
8470                            0              1            0  
8471                            0              0          

In [13]:
# This is a test cell
# If **NO** message is printed, it means that the tests passed
# One basic test to see if your code works
assert_equal(hotel_coded.columns[12], 'gender_Female', msg='You did not create the dummy variables correctly')
assert_equal(hotel_coded.columns[13], 'gender_Male', msg='You did not create the dummy variables correctly')

<hr>

## Step 4
Drop the column representing the non-member tier of the loyalty program and the column representing females. Be sure to keep your newly trimmed data in the variable `hotel_coded`. 

In [14]:
# YOUR CODE HERE
# view col names
hotel_coded.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8472 entries, 0 to 8471
Data columns (total 14 columns):
 #   Column                           Non-Null Count  Dtype         
---  ------                           --------------  -----         
 0   record_id                        8472 non-null   int64         
 1   arrival_date                     8472 non-null   datetime64[ns]
 2   age                              8472 non-null   int64         
 3   maintenance_and_upkeep           8472 non-null   int64         
 4   staff_service_overall            8472 non-null   int64         
 5   overall_satisfaction             8472 non-null   int64         
 6   intent_to_recommend              8472 non-null   int64         
 7   loyalty_program_tier_Basic       8472 non-null   int32         
 8   loyalty_program_tier_Gold        8472 non-null   int32         
 9   loyalty_program_tier_Non-Member  8472 non-null   int32         
 10  loyalty_program_tier_Platinum    8472 non-null   int32      

In [15]:
# drop loyalty_program_tier_Non-Member and gender_Female
hotel_coded = hotel_coded.drop(columns=['loyalty_program_tier_Non-Member' , 'gender_Female'])

In [16]:
# confirm dropped
hotel_coded.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8472 entries, 0 to 8471
Data columns (total 12 columns):
 #   Column                         Non-Null Count  Dtype         
---  ------                         --------------  -----         
 0   record_id                      8472 non-null   int64         
 1   arrival_date                   8472 non-null   datetime64[ns]
 2   age                            8472 non-null   int64         
 3   maintenance_and_upkeep         8472 non-null   int64         
 4   staff_service_overall          8472 non-null   int64         
 5   overall_satisfaction           8472 non-null   int64         
 6   intent_to_recommend            8472 non-null   int64         
 7   loyalty_program_tier_Basic     8472 non-null   int32         
 8   loyalty_program_tier_Gold      8472 non-null   int32         
 9   loyalty_program_tier_Platinum  8472 non-null   int32         
 10  loyalty_program_tier_Silver    8472 non-null   int32         
 11  gender_Male      

<hr>

## Step 5
Recall your overall goal is to predict a patron's response to the question regarding their intent to recommend the hotel to their friends. Create a variable named `y` from the intent to recommend column. Create a variable named `X` that contains all of the other columns **except `record_id`, `arrival_date`, and `intent_to_recommend`**. (Thought Exercise: Why shouldn't you include those variables as part of `X`?) 

In [17]:
# YOUR CODE HERE
y = hotel_coded.intent_to_recommend
X = hotel_coded.drop(columns=['record_id', 'arrival_date', 'intent_to_recommend'])

In [18]:
# view y
y.info()

<class 'pandas.core.series.Series'>
RangeIndex: 8472 entries, 0 to 8471
Series name: intent_to_recommend
Non-Null Count  Dtype
--------------  -----
8472 non-null   int64
dtypes: int64(1)
memory usage: 66.3 KB


In [19]:
# view X
X.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8472 entries, 0 to 8471
Data columns (total 9 columns):
 #   Column                         Non-Null Count  Dtype
---  ------                         --------------  -----
 0   age                            8472 non-null   int64
 1   maintenance_and_upkeep         8472 non-null   int64
 2   staff_service_overall          8472 non-null   int64
 3   overall_satisfaction           8472 non-null   int64
 4   loyalty_program_tier_Basic     8472 non-null   int32
 5   loyalty_program_tier_Gold      8472 non-null   int32
 6   loyalty_program_tier_Platinum  8472 non-null   int32
 7   loyalty_program_tier_Silver    8472 non-null   int32
 8   gender_Male                    8472 non-null   int32
dtypes: int32(5), int64(4)
memory usage: 430.3 KB


<hr>

## Step 6
Split your data into training and test sets. You should use a training set size of 75% of the data and a random seed of 101. You should use the commonly used names of `y_train`, `y_test`, `X_train`, and `X_test` when splitting the data.

In [20]:
# YOUR CODE HERE
# import package to split the data
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.25, random_state=101)

In [21]:
# This is a test cell
# If **NO** message is printed, it means that the tests passed
# One basic test to see if your code works
assert_equal(type(y_test), pd.Series, msg='You did not create y_test correctly')

<hr>

## Step 7
Use the `LinearRegression` class from scikit-learn to fit an ordinary least squares model on the **training dataset**. What is the estimated coefficient on the variable representing males? Store the value, **without rounding**, in a variable named `coef_males`.

In [22]:
# YOUR CODE HERE
# bring in LinearRegression package
from sklearn.linear_model import LinearRegression

reg = LinearRegression()
reg.fit(X_train, y_train)

# display coefficients
reg.coef_

array([-0.00126319,  0.01454387, -0.05843249,  0.85475996,  0.05356864,
        0.06871536,  0.10001063,  0.09543392, -0.02106501])

In [23]:
# make into coefficients DataFrame
coefficients_df = pd.DataFrame(
data = reg.coef_,
index = X_train.columns,
columns = ['Coefficient Value'])

coefficients_df

Coefficient Value
age                                    -0.001263
maintenance_and_upkeep                  0.014544
staff_service_overall                  -0.058432
overall_satisfaction                    0.854760
loyalty_program_tier_Basic              0.053569
loyalty_program_tier_Gold               0.068715
loyalty_program_tier_Platinum           0.100011
loyalty_program_tier_Silver             0.095434
gender_Male                            -0.021065

In [24]:
# pull out the estimated coefficient on the variable representing males (gender_Male) without rounding
float(coefficients_df.values[8])

-0.02106501331610483

In [25]:
# store as coef_males
coef_males = float(coefficients_df.values[8])

In [26]:
# confirm not rounded and number dtype
print(coef_males)
print(type(coef_males))

-0.02106501331610483
<class 'float'>


In [27]:
# This is a test cell
# If **NO** message is printed, it means that the tests passed
# One basic test to see if your code works
assert_in('coef_males', dir(), msg='You did not name your variable `coef_males` as instructed')

<hr>

## Step 8
Calculate the $R^2$, MSE, and RMSE on the **training** set. Store them in the variables named `train_R2`, `train_MSE`, and `train_RMSE`.

In [28]:
# YOUR CODE HERE
from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score
from sklearn.metrics import mean_squared_error

# R-Squared (R^2)
train_R2 = r2_score(y_train, reg.predict(X_train))
train_R2

0.5991001853064454

In [29]:
# Mean Squared Error (MSE)
train_MSE = mean_squared_error(y_train, reg.predict(X_train))
train_MSE

1.8301145300801605

In [30]:
# Root Mean Squared Error
import numpy as np
train_RMSE = np.sqrt(train_MSE)
train_RMSE

1.3528172567202712

In [31]:
# This is a test cell
# If **NO** message is printed, it means that the tests passed
# One basic test to see if your code works
assert_in('train_R2', dir(), msg='You did not name your variable `train_R2` as instructed')
assert_in('train_MSE', dir(), msg='You did not name your variable `train_MSE` as instructed')
assert_in('train_RMSE', dir(), msg='You did not name your variable `train_RMSE` as instructed')

<hr>

## Step 9
Using your fitted model, predict the intent to recommend for the **test** set. Calculate the $R^2$, MSE, and RMSE on the **test** set. Store them in the variables named `test_R2`, `test_MSE`, and `test_RMSE`. 

In [32]:
# YOUR CODE HERE
# create predication variable 'pred' for predictions
pred = reg.predict(X_test)

In [33]:
# R-squared
test_R2 = r2_score(y_test, pred)
test_R2

0.5932229411830202

In [34]:
# Mean Squared Error
test_MSE = mean_squared_error(y_test, pred)
test_MSE

1.892848692015891

In [35]:
# Root Mean Squared Error
test_RMSE = np.sqrt(test_MSE)
test_RMSE

1.3758083776514414

In [36]:
# This is a test cell
# If **NO** message is printed, it means that the tests passed
# One basic test to see if your code works
assert_in('test_R2', dir(), msg='You did not name your variable `test_R2` as instructed')
assert_in('test_MSE', dir(), msg='You did not name your variable `test_MSE` as instructed')
assert_in('test_RMSE', dir(), msg='You did not name your variable `test_RMSE` as instructed')

<hr>

## Step 10
Using the `statsmodels` package, fit an ordinary least squares model to the training data and examine the results. Store the results in the variable `mlr_results` and print it out. Take a look at the $p$-values for the independent variables. 

In [37]:
# YOUR CODE HERE
# import ols package
from statsmodels.formula.api import ols

# copy X_train
data_full = X_train.copy()

# add col called 'intent_to_recommend' from y_train
data_full['intent_to_recommend'] = y_train

# create the RHS of all the X variables
rhs = '+'.join(X_train.columns) # join all the columns in X_train with a plus sign

# run ols
mlr_results = ols('intent_to_recommend ~'+rhs, data=data_full).fit()

In [38]:
# view
mlr_results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                             OLS Regression Results                            
===============================================================================
Dep. Variable:     intent_to_recommend   R-squared:                       0.599
Model:                             OLS   Adj. R-squared:                  0.599
Method:                  Least Squares   F-statistic:                     1053.
Date:                 Mon, 11 Dec 2023   Prob (F-statistic):               0.00
Time:                         12:06:17   Log-Likelihood:                -10936.
No. Observations:                 6354   AIC:                         2.189e+04
Df Residuals:                     6344   BIC:                         2.196e+04
Df Model:                            9                                         
Covariance Type:             nonrobust                                         
=================================================================================================
                                    coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------------------
Intercept                         1.0931      0.111      9.815      0.000       0.875       1.311
age                              -0.0013      0.001     -1.149      0.251      -0.003       0.001
maintenance_and_upkeep            0.0145      0.016      0.922      0.357      -0.016       0.045
staff_service_overall            -0.0584      0.019     -3.106      0.002      -0.095      -0.022
overall_satisfaction              0.8548      0.013     65.208      0.000       0.829       0.880
loyalty_program_tier_Basic        0.0536      0.038      1.398      0.162      -0.022       0.129
loyalty_program_tier_Gold         0.0687      0.075      0.918      0.359      -0.078       0.215
loyalty_program_tier_Platinum     0.1000      0.072      1.386      0.166      -0.041       0.241
loyalty_program_tier_Silver       0.0954      0.066      1.449      0.147      -0.034       0.225
gender_Male                      -0.0211      0.034     -0.619      0.536      -0.088       0.046
==============================================================================
Omnibus:                      210.624   Durbin-Watson:                   2.000
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              117.720
Skew:                          -0.168   Prob(JB):                     2.74e-26
Kurtosis:                       2.424   Cond. No.                         424.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [39]:
print(type(mlr_results))

<class 'statsmodels.regression.linear_model.RegressionResultsWrapper'>


In [40]:
# p-values <0.05: staff_service_overall, overall_satisfaction

In [41]:
# This is a test cell
# If **NO** message is printed, it means that the tests passed
# One basic test to see if your code works
assert_in('mlr_results', dir(), msg='You did not name the variable `mlr_results` as instructed')

<hr>

## Step 11
Using only variables that had a $p$-value < 0.05, fit a new ordinary least squares model to the training data and examine the results. Store these new results in variable named `second_mlr_results` and print it out.

In [42]:
# YOUR CODE HERE
# remove all but staff_service_overall and overall_satisfaction first
data_full_second = data_full.drop(columns=['age', 'maintenance_and_upkeep', 'loyalty_program_tier_Basic',
                                          'loyalty_program_tier_Gold', 'loyalty_program_tier_Platinum', 
                                          'loyalty_program_tier_Silver', 'gender_Male'])

In [43]:
# confirm
data_full_second

staff_service_overall  overall_satisfaction  intent_to_recommend
1962                      8                     8                    7
3827                      9                     8                    5
7235                      8                     9                   10
1672                      8                     7                    5
1221                      1                     2                    1
...                     ...                   ...                  ...
599                       5                     3                    7
5695                      8                     5                    5
8006                      7                     7                    7
1361                      9                     9                    9
1547                      8                     6                    5

[6354 rows x 3 columns]

In [44]:
# create the RHS of all the remaining X variables
## Town Hall Question - in join(used data_full_second instead of X because no longer need old cols) - correct way?
rhs_second = '+'.join(data_full_second.columns.drop('intent_to_recommend')) # join all the columns in X_train with a plus sign

# run ols
second_mlr_results = ols('intent_to_recommend ~'+rhs_second, data=data_full_second).fit()

In [45]:
# view
second_mlr_results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                             OLS Regression Results                            
===============================================================================
Dep. Variable:     intent_to_recommend   R-squared:                       0.599
Model:                             OLS   Adj. R-squared:                  0.599
Method:                  Least Squares   F-statistic:                     4737.
Date:                 Mon, 11 Dec 2023   Prob (F-statistic):               0.00
Time:                         12:06:17   Log-Likelihood:                -10940.
No. Observations:                 6354   AIC:                         2.189e+04
Df Residuals:                     6351   BIC:                         2.191e+04
Df Model:                            2                                         
Covariance Type:             nonrobust                                         
=========================================================================================
                            coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------------
Intercept                 1.0770      0.076     14.081      0.000       0.927       1.227
staff_service_overall    -0.0469      0.014     -3.320      0.001      -0.075      -0.019
overall_satisfaction      0.8546      0.013     65.249      0.000       0.829       0.880
==============================================================================
Omnibus:                      208.158   Durbin-Watson:                   1.999
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              116.945
Skew:                          -0.168   Prob(JB):                     4.03e-26
Kurtosis:                       2.427   Cond. No.                         51.7
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [46]:
# This is a test cell
# If **NO** message is printed, it means that the tests passed
# One basic test to see if your code works
assert_in('second_mlr_results', dir(), msg='You did not name the variable `second_mlr_results` as instructed')

<hr>

## Final Step
Compare the results you found in Step 10 and Step 11. In the markdown cell below, carefully explain which model you would use and and why.

In Step 10, I had an R-Squared value of: 0.599 and Adjusted R-Squared value of: 0.599 as well. This indicates that the model cuirrently explains approximately 59.9% of the variance with the X-variables of age, maintenance_upkeep, staff_service_overall, overall_satisfaction, loyalty_program_tier_Basic, loyalty_program_tier_Gold, loyalty_program_tier_Platinum, loyalty_program_tier_Silver, and gender_Male. There were two statistically significant variables from Step 10: staff_service_overall and overall_satisfaction as the p-values were <0.05 and overall_satisfaction was actually statistically significiant at the P<.001 level. 

In Step 11, I had an R-Squared value of: 0.599 and Adjusted R-Squared value of: 0.599 as well. This indicates thatt the model currently explains 59.9% of the variance with the X-variables of staff_service_overall and overall_satisfaction. To further indicate the best model (if historical context required) we could remove the least significant variables first (highest p-values) until the model potentially remained with only statistically significant variables at the p<.05 level. Therefore, I would remain with the Step 10 model because there may be variables that become statistically significant if we continue to eliminate the least significant variables first until all  variables are significant. 

**&copy; 2022 - Present: Matthew D. Dean, Ph.D.   
Clinical Associate Professor of Business Analytics at William \& Mary.**